src/strings-in-block/dom-handler/dom-handle.php

```diff
@@ -121,7 +121,13 @@ private function cloneNodeWithoutChildren( \DOMNode $element ) {
 	}
 
 	protected function getAsHTML5( \DOMNode $element ) {
-		return strtr( $element->ownerDocument->saveXML( $element, LIBXML_NOEMPTYTAG ),
+		$html = str_replace(
```

Comment:
I think this is due to the recent changes. When we use `saveXML`, it wraps everything inside `<body>`. So I had to remove it.

Reply:
@strategio Are you sure about this. Wouldn't everything else be broken if this code was wrapping in a body tag.

Reply:
@BruceOnTheGo @strategio 

I think the problem is when we load the HTML (I assume it's in `\WPML\PB\Gutenberg\StringsInBlock\DOMHandler\DOMHandle::getDom`).

I did a quick test with this snippet:

```php
$html = <<<'HTML'
<form action="http://maps.google.com/maps" method="get" target="_blank">
   <label for="saddr">Enter your location</label>
   <input type="text" name="saddr" />
   <input type="hidden" name="daddr" value="350 5th Ave New York, NY 10018 (Empire State Building)" />
   <input type="submit" value="Get directions" />
</form>
HTML;


$dom = new \DOMDocument();
$dom->loadHTML( $html );
//$dom->loadHTML( $html, LIBXML_HTML_NOIMPLIED );

$children = $dom->childNodes;

foreach ( $children as $child ) {
	echo $child->ownerDocument->saveXML( $child, LIBXML_NOEMPTYTAG );
}
```

I doesn't only add the `<body>` tag. In my case, I also get `<html>`.

Now, if you comment out `$dom->loadHTML( $html );` and uncomment `$dom->loadHTML( $html, LIBXML_HTML_NOIMPLIED );`, no extra tags are added.

Reply:
We had some magic in `DOMHandle::getDom` to handle this. But with the suggestion from @andreasciamanna it can probably be simplified.

Reply:
Initially I was not using  `DOMHandle::getDom` in my new code. Now this is done and I don't need the change here anymore.

I applied Andrea' suggestion and it seems to do the trick.